src/Compilers/Core/Portable/Syntax/SyntaxTokenList.cs

```diff
@@ -77,16 +74,21 @@ public static SyntaxTokenList Create(ReadOnlySpan<SyntaxToken> tokens)
             if (tokens == null)
                 return null;
 
-            // TODO: it would be nice to avoid the intermediary builder.  We could just inline what ToList does here.
-            var builder = new SyntaxTokenListBuilder(tokens.Length);
-            for (int i = 0; i < tokens.Length; i++)
+            switch (tokens.Length)
             {
-                var node = tokens[i].Node;
-                Debug.Assert(node is not null);
-                builder.Add(node);
-            }
+                case 0: return null;
+                case 1: return tokens[0].Node;
+                case 2: return Syntax.InternalSyntax.SyntaxList.List(tokens[0].Node!, tokens[1].Node!);
+                case 3: return Syntax.InternalSyntax.SyntaxList.List(tokens[0].Node!, tokens[1].Node!, tokens[2].Node!);
+                default:
+                    {
+                        var copy = new GreenNode[tokens.Length];
+                        for (int i = 0, n = tokens.Length; i < n; i++)
+                            copy[i] = tokens[i].Node!;
 
-            return builder.ToList().Node;
+                        return Syntax.InternalSyntax.SyntaxList.List(copy, tokens.Length);
```

Comment:
yeah.  it's very weird, and i don't remember the historical context.  my recollection is that ArrayElement was supposed to be better because it removes variance checks in array accesses.  but i don't know why we're not uniform in this.  i would expect we would either use it in all cases, or none.

Reply:
actually, you found something interesting.  we were doing an additional extra copy.  you helped find that, which i've now removed :)